ganttproject/src/net/sourceforge/ganttproject/export/ExporterToCSV.java

```diff
@@ -18,27 +18,76 @@
  */
 package net.sourceforge.ganttproject.export;
 
-import java.awt.Component;
+import biz.ganttproject.core.option.EnumerationOption;
+import biz.ganttproject.core.option.GPAbstractOption;
+import biz.ganttproject.core.option.GPOption;
+import biz.ganttproject.core.option.GPOptionGroup;
+import biz.ganttproject.impex.csv.GanttCSVExport;
+import net.sourceforge.ganttproject.GPLogger;
+import net.sourceforge.ganttproject.GanttProject;
+import net.sourceforge.ganttproject.io.CSVOptions;
+import net.sourceforge.ganttproject.language.GanttLanguage;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+
+import java.awt.*;
 import java.io.BufferedOutputStream;
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.OutputStream;
+import java.util.Arrays;
 import java.util.List;
 
-import net.sourceforge.ganttproject.GPLogger;
-import net.sourceforge.ganttproject.GanttProject;
-import net.sourceforge.ganttproject.language.GanttLanguage;
+public class ExporterToCSV extends ExporterBase {
+  static class FileTypeOption extends GPAbstractOption<String> implements EnumerationOption {
+    static final String[] FILE_FORMAT_IDS = new String[]{"impex.csv.fileformat.csv", "impex.csv.fileformat.xls"};
```

Comment:
Just've realized that we can do this code simpler as well (sorry that didn't write it earlier). 

Look at `ImportCalendarOption` class -- it implements enumeration option based on Java enum values. I think it makes sense to do the same here. You will get option.getSelectedValue() returning your enum and will be able to do something like

```
String proposeFileExtension() {
   return getSelectedValue().getExtension();
}
```
instead of all those loops.